--- ldlm_lockd.c.orig	2014-09-15 12:18:57.671454972 +0000
+++ ldlm_lockd.c	2014-09-15 12:19:53.747456921 +0000
@@ -2529,7 +2529,11 @@
 	bltd.bltd_num = cfs_atomic_read(&blp->blp_num_threads);
 	snprintf(bltd.bltd_name, sizeof(bltd.bltd_name) - 1,
 		"ldlm_bl_%02d", bltd.bltd_num);
+        #pragma GCC diagnostic push
+        #pragma GCC diagnostic ignored "-Wformat-security"
 	task = kthread_run(ldlm_bl_thread_main, &bltd, bltd.bltd_name);
+        #pragma GCC diagnostic pop
+
 	if (IS_ERR(task)) {
 		CERROR("cannot start LDLM thread ldlm_bl_%02d: rc %ld\n",
 		       cfs_atomic_read(&blp->blp_num_threads), PTR_ERR(task));
