--- qsd_reint.c.orig	2014-09-15 13:57:11.980882845 +0000
+++ qsd_reint.c	2014-09-15 13:57:49.268884141 +0000
@@ -664,7 +664,10 @@
 	snprintf(name, MTI_NAME_MAXLEN, "qsd_reint_%d.%s",
 		 qqi->qqi_qtype, qsd->qsd_svname);
 
+        #pragma GCC diagnostic push
+        #pragma GCC diagnostic ignored "-Wformat-security"
 	rc = PTR_ERR(kthread_run(qsd_reint_main, (void *)qqi, name));
+        #pragma GCC diagnostic pop
 	OBD_FREE(name, MTI_NAME_MAXLEN);
 
 	if (IS_ERR_VALUE(rc)) {
