--- pinger.c	2014-09-17 10:10:54.530025420 +0000
+++ pinger.c.orig	2014-09-17 10:09:46.346026812 +0000
@@ -334,11 +334,8 @@
 
 	/* CLONE_VM and CLONE_FILES just avoid a needless copy, because we
 	 * just drop the VM and FILES in kthread_run() right away. */
-        #pragma GCC diagnostic push
-        #pragma GCC diagnostic ignored "-Wformat-security"
 	rc = PTR_ERR(kthread_run(ptlrpc_pinger_main,
 				 &pinger_thread, pinger_thread.t_name));
-        #pragma GCC diagnostic pop
 	if (IS_ERR_VALUE(rc)) {
 		CERROR("cannot start thread: %d\n", rc);
 		RETURN(rc);
